Tests/Common/SimulatedDevice.cs

```diff
@@ -226,50 +251,72 @@ private bool HandleJoinAccept(LoRaPayloadJoinAccept payload)
             return true;
         }
 
+        //// Sends unconfirmed message
+        public Task SendDataMessageAsync(LoRaRequest loRaRequest) =>
+            Task.WhenAll(from basicsStation in this.simulatedBasicsStations
+                         select basicsStation.SendDataMessageAsync(loRaRequest, CancellationToken.None));
+
+        public void EnsureMessageResponsesAreReceived(int expectedCout)
+        {
+            Assert.Equal(expectedCout, this.receivedMessages.Count);
+        }
+
         // Performs join
-        public async Task<bool> JoinAsync(LoRaRequest joinRequest, SimulatedBasicsStation basicsStation, TimeSpan? timeout = null)
+        public async Task<bool> JoinAsync(TimeSpan? timeout = null)
         {
             timeout ??= TimeSpan.FromSeconds(30);
             using var joinCompleted = new SemaphoreSlim(0);
+            using var joinRequest = WaitableLoRaRequest.CreateWaitableRequest(CreateJoinRequest());
             var joinRequestPayload = (LoRaPayloadJoinRequest)joinRequest.Payload;
             var joinSuccessful = false;
 
             void OnMessageReceived(object sender, EventArgs<string> response)
             {
-                // handle join
-                var joinAcceptstring = JsonSerializer.Deserialize<JoinAcceptResponse>(response.Value);
-                // is null in case it is another message coming from the station.
-                if (joinAcceptstring?.Pdu != null)
+                try
                 {
-                    var joinAccept = new LoRaPayloadJoinAccept(ConversionHelper.StringToByteArray(joinAcceptstring.Pdu), AppKey.Value);
-
-                    var result = HandleJoinAccept(joinAccept); // may need to return bool and only release if true.
-                    joinSuccessful = result;
+                    var joinAcceptResponse = JsonSerializer.Deserialize<JoinAcceptResponse>(response.Value);
+                    // PDU is null in case it is another message coming from the station.
+                    if (joinAcceptResponse is { } someJoinAcceptResponse && someJoinAcceptResponse.Pdu is { } somePdu && someJoinAcceptResponse.DevEui == DevEUI)
+                    {
+                        var joinAccept = new LoRaPayloadJoinAccept(ConversionHelper.StringToByteArray(somePdu), AppKey.Value);
+                        joinSuccessful = HandleJoinAccept(joinAccept); // may need to return bool and only release if true.
```

Comment:
What about this comment?

Reply:
good catch, done in d71a6be4c05e105cc3c92f3efd5a2076325da6e6